components/brave_wallet/browser/zcash/rust/orchard_decoded_blocks_bundle_impl.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_RUST_ORCHARD_DECODED_BLOCKS_BUNDLE_IMPL_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_RUST_ORCHARD_DECODED_BLOCKS_BUNDLE_IMPL_H_
+
+#include <vector>
+
+#include "brave/components/brave_wallet/browser/zcash/rust/lib.rs.h"
+#include "brave/components/brave_wallet/browser/zcash/rust/orchard_decoded_blocks_bundle.h"
+#include "third_party/rust/cxx/v1/cxx.h"
+
+namespace brave_wallet::orchard {
+
+class OrchardDecodedBlocksBundleImpl : public OrchardDecodedBlocksBundle {
```

Comment:
Is there a particular reason why we are separating `OrchardDecodedBlocksBundle` and `OrchardDecodedBlocksBundleImpl`?

Reply:
OrchardDecodedBlocksBundleImpl contains rust headers, policy is not to expose rust headers to the non-rust code